### ISSUE TITLE
Feature Request: Elastic Beanstalk Application Data Source

### ISSUE THREAD

**Issue by Sneagan** [terraform-providers/terraform-provider-aws#3838]
### Terraform Version
0.11.3

### Affected Resource(s)
- aws_elastic_beanstalk_application

### Proposed Usage
```hcl
# Beanstalk Application
// To create:
// resource "aws_elastic_beanstalk_application" "default" {
//   name        = "${var.application_name}"
//   description = "${var.application_description}"
// }

// To Reference:
data "aws_elastic_beanstalk_application" "default" {
  name = "${var.application_name}"
}
```

**Comment by bflad** [comment]
A new `aws_elastic_beanstalk_application` data source has been merged and will release with version 1.56.0 of the Terraform AWS provider, likely in the next day or two.

**Comment by bflad** [comment]
This has been released in [version 1.56.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1560-january-16-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
